compiler/rustc_builtin_macros/src/test_harness.rs

```diff
@@ -129,8 +130,8 @@ impl<'a> MutVisitor for TestHarnessGenerator<'a> {
         c.items.push(mk_main(&mut self.cx));
     }
 
-    fn flat_map_item(&mut self, mut i: P<ast::Item>) -> SmallVec<[P<ast::Item>; 1]> {
-        let item = &mut *i;
+    fn visit_item(&mut self, i: &mut P<ast::Item>) {
+        let item = i.deref_mut();
```

Comment:
Could you pick up the test harness cleanups from https://github.com/rust-lang/rust/pull/127371?
The `deref_mut`s shouldn't be necessary in particular.

Reply:
There is one place where an intermediate unwrapped (`&mut Item` vs `&mut P<Item>`) is still needed. There we need multiple mut refs to fields of `Item` to pass to `walk_item_kind`. I swapped it to `&mut **item` so we don't include `DerefMut`